### ISSUE TITLE
Compiling and using Liftoff projects as 3rd party libraries

### ISSUE THREAD

**Issue by Mr00Anderson** [tommyettinger/gdx-liftoff#26]
This is to record discord events regarding an issue with wanting to use liftoff projects as a dependency such as a API jar. This does not work out of box because the `gwt.xml` files are not kept through compiling when using maven/java to produce a artifact. In order to get the `.gwt.xml` to stay through compilation the only solution that worked was ```    jar {
        from project.sourceSets.main.allSource
        from project.sourceSets.main.output
    }``` to be used in the module that you want to publish on maven. This will include the `gwt.xml` in the source jar output on maven.

**Comment by Mr00Anderson** [comment]
Solution was found here. https://stackoverflow.com/questions/28458016/gradle-generated-jar-does-not-include-gwt-xml-file

**Comment by tommyettinger** [comment]
Are you referring to Maven, the build tool, or Maven Central, the repository that Gradle can use? Maven (build tool) issues aren't applicable to Gradle projects.

Liftoff isn't designed for libraries; it's meant for applications. I usually use one of my existing libraries as a template when I make a new one; the original Gradle-based library I used as a basis was typing-label by RafaSKB. With the template approach, I don't involve Liftoff unless it's to add the library, eventually, to the third-party extensions. If you want a "template library" that doesn't depend on libGDX, [jdkgdxds](https://github.com/tommyettinger/jdkgdxds) is small enough; if it should depend on libGDX, [anim8-gdx](https://github.com/tommyettinger/anim8-gdx) is a good start (if you remove the big images/ folder). Both of those use src/main/resources/ to hold the .gwt.xml file. There's also still typing-label as a good option.

Closing as out of scope; if there's a case for using an application as a library (on GWT) I can reopen.
